extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BuildTimeEnabledProcessor.java

```diff
@@ -72,7 +72,7 @@ void unlessBuildProfile(CombinedIndexBuildItem index, BuildProducer<BuildTimeCon
     }
 
     @BuildStep
-    void ifBuildProperty(BeanArchiveIndexBuildItem index, BuildProducer<BuildTimeConditionBuildItem> producer) {
+    void ifBuildProperty(CombinedIndexBuildItem index, BuildProducer<BuildTimeConditionBuildItem> producer) {
```

Comment:
Use a `CombinedIndexBuildItem` instead of a `BeanArchiveIndexBuildItem` to avoid getting the next `ChainBuildException`:
```
java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Cycle detected:
       io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveCDIProcessor#additionalBeans produced class io.quarkus.arc.deployment.AdditionalBeanBuildItem
    to io.quarkus.arc.deployment.BeanArchiveProcessor#build produced class io.quarkus.arc.deployment.BeanArchiveIndexBuildItem
    to io.quarkus.arc.deployment.BuildTimeEnabledProcessor#unlessBuildProperty produced class io.quarkus.arc.deployment.BuildTimeConditionBuildItem
    to io.quarkus.resteasy.reactive.common.deployment.ResteasyReactiveCommonProcessor#handleApplication produced class io.quarkus.resteasy.reactive.common.deployment.ApplicationResultBuildItem
    to io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveScanningProcessor#scanForFeatures produced class io.quarkus.resteasy.reactive.spi.JaxrsFeatureBuildItem
    to io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveCDIProcessor#additionalBeans
```

Reply:
Which change made in this PR necessitated this?

Reply:
@geoand It's needed because of [this change](https://github.com/quarkusio/quarkus/pull/15215/files#diff-0b225111895e58c28abd2252e84580d5bd94b727c4d88687ae3c9f318f1e5d3cR79) that makes `ResteasyReactiveCommonProcessor#handleApplication` depends on `BuildTimeConditionBuildItem` which finally causes a cycle as we can see in the stack trace

Reply:
I am not a fan of this change, but we can probably live with it if we don't find a better way

Reply:
@geoand I agree, I don't like it too but I could not find any better way. I would like to know why properties are managed with a `BeanArchiveIndexBuildItem` while profiles are managed with `CombinedIndexBuildItem`?

Reply:
I added another comment about this.

We'll need to change this as it could break existing code.

Reply:
I'd argue that using `CombinedIndexBuildItem` in profiles is wrong as well and was probably done by mistake...

Reply:
@geoand fixed with the new build item `EagerBuildTimeConditionBuildItem`